website/docs/guides/vm_guest_customization.html.markdown

```diff
@@ -105,3 +105,63 @@ After applying the configuration one should be able to login using `core` user a
 password. In addition the hostname should be set to `core1. [Official Ignition
 docs](https://docs.fedoraproject.org/en-US/fedora-coreos/producing-ign/) have a lot more
 configuration options.
+
+## Guest Customization using CloudInit (OVF datasource)
+
+Configuration can be passed to CloudInit using
+[`guest_properties`](/providers/vmware/vcd/latest/docs/resources/vapp_vm#guest_properties) key value
+map in [`vcd_vapp_vm`](/providers/vmware/vcd/latest/docs/resources/vapp_vm) or
+[`vcd_vm`](/providers/vmware/vcd/latest/docs/resources/vm) resources.
+
+More about [CloudInit](https://cloudinit.readthedocs.io/en/latest/) and
+[OVF](https://cloudinit.readthedocs.io/en/latest/topics/datasources/ovf.html)
+[datasource](https://cloudinit.readthedocs.io/en/latest/topics/datasources.html)
+
+### Example using CloudInit configuration (Ubuntu 22.04 LTS)
+
+[Ubuntu 22.04 LTS OVA
+Image](https://cloud-images.ubuntu.com/releases/22.04/release/ubuntu-22.04-server-cloudimg-amd64.ova)
+from [Ubuntu Cloud images repository](https://cloud-images.ubuntu.com/) used.
+
+Prepare a script to run using Cloud Init. In this example it is just a PoC (note the `sudo` command
+to run as `root` as by default commands are run as `ubuntu` user).
+
+Store it in `script.sh` (or change user-data references in HCL script). Shebang `#!/bin/sh` is a
+required so that CloudInit recognizes it is a shell script. There are [other possible
+formats](https://cloudinit.readthedocs.io/en/latest/topics/format.html#user-data-script) for
+supplying CloudInit user data.
+
+```sh
+#!/bin/sh
+echo "Command executed as 'ubuntu' user" >> /tmp/setup.log
+sudo echo "Command executed as 'root' user" >> /tmp/setup.log
+```
+
+VM definition might be different, but key here is `guest_properties`. It will supply needed data for
+CloudInit (including the shell script in file `script.sh`)
```

Comment:
Additionally, maybe add a sentence about using base64encode?

Reply:
Additionally mentioned base64 encoded file is requirement by CloudInit